### PR TITLE
Add CORS HTTP headers to regular (non-OPTIONS) requests

### PR DESCRIPTION
**Pull request by or-else**
Ref https://github.com/tinode/chat/pull/906

### CODE REVIEW

**Comment by or-else** [comment]
@markeloff86 please check if it solves your problem.

**Comment by markeloff86** [comment]
So now we are adding Access-Control-Allow-Origin for all method?
I think It might work

File: server/media/media.go [hunk 1]
```diff
@@ -88,49 +89,59 @@ func matchCORSMethod(allowMethods []string, method string) bool {
...
+			allowMethods = []string{http.MethodPost, http.MethodPut, http.MethodHead, http.MethodOptions}
+		}
+
+		if !matchCORSMethod(allowMethods, requestMethod) {
+			// CORS policy does not allow this method.
+			return headers, http.StatusNoContent
+		}
+
+		headers["Access-Control-Allow-Headers"] = []string{"*"}
+		headers["Access-Control-Allow-Credentials"] = []string{"true"}
```
Comment (markeloff86) [inline server/media/media.go:131]: array?
The Access-Control-Allow-Credentials header expects a single string value, not a slice of strings

**Comment by or-else** [comment]
> But I'm not sure about Access-Control-Allow-Credentials Should we add it too for all methods? Because we use auth headers https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials

The way I read the spec it's not needed if request is preflighted and reflight includes `Access-Control-Allow-Credentials`:

* If the request is preflighted [... and] the server's response to the preflight request sets the Access-Control-Allow-Credentials header to true, then the real request will include credentials...

**Comment by or-else** [comment]
Could you please just test it without your setup?

**Comment by markeloff86** [comment]
Your new code doesn't add headers for POST and PUT
So I've made some changes and tested it
This works for me
https://github.com/tinode/chat/pull/909

**Comment by or-else** [comment]
> it crashes here

How does it crash? Do you have a stack trace?

**Comment by or-else** [comment]
> Your new code doesn't add headers for POST and PUT, because it crashes here
> 
> https://github.com/tinode/chat/blob/99c633297a805123f74642334f87d4a5b8ab7967/server/media/s3/s3.go#L164
> 
> 
> So I've made some changes and tested it
> This works for me
> #909

I don't see how it could possibly crash. I can see how it could return an error. Could you please elaborate.

**Comment by markeloff86** [comment]
I'm not sure that's exacly problem in this line of code, but without condition which I've added for POST and PUT I always get 404 status code without Access-Control-Allow-Origin header.
So I presume 404 status code return this line of code https://github.com/tinode/chat/blob/99c633297a805123f74642334f87d4a5b8ab7967/server/media/s3/s3.go#L166

it returns because for POST and PUT we don't pass id. That's why this code shouldn't have been executed for POST and PUT.

**Comment by or-else** [comment]
OK, that's what I thought. It's not a crash, but an error.

**Pull request merged**